### PR TITLE
Add org.kde.index

### PR DESCRIPTION
**Pull request by ognarb**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. (I contribute a bit to Maui apps but I'm not the maintainer. The maintainer knows that I'm submitting this MR)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). (sort of but it's part of KDE and I'm a KDE dev)
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by ognarb** [comment]
bot, build org.kde.index

**Comment by ognarb** [comment]
bot, build org.kde.index

**Comment by ognarb** [comment]
bot, build org.kde.index

**Comment by ognarb** [comment]
bot, build org.kde.index

File: org.kde.index.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "index",
+    "tags": ["nightly"],
+    "desktop-file-name-suffix": " (Nightly)",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
```
Comment (hfiguiere) [inline org.kde.index.json:13]:
```suggestion
        "--socket=fallback-x11",
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.index.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "index",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--filesystem=host",
+        "--env=XDG_DATA_DIRS=/usr/share:/app/share/"
```
Comment (hfiguiere) [inline org.kde.index.json:15]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.index.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    ],
+    "separate-locales": false,
```
Comment (barthalion) [inline org.kde.index.json:17]:
```suggestion
```

**Comment by ognarb** [comment]
bot, build org.kde.index

**Comment by nedrichards** [comment]
@flathub/kde FYI

**Comment by travier** [comment]
Not tested but LGTM. Have you submitted the metadat patches upstream? Thanks

**Pull request closed without merging**